modules/dynamodb/dynamodb_test.go

```diff
@@ -0,0 +1,258 @@
+	testcontainers.CleanupContainer(t, ctr2)
+	require.NoError(t, err)
+
+	// fetch client handle again
+	cli2, err := getDynamoDBClient(ctr2)
+	require.NoError(t, err, "failed to get dynamodb client handle")
+
+	// list tables and verify
+
+	result, err := cli2.ListTables(context.Background(), nil)
+	require.NoError(t, err, "dynamodb list tables operation failed")
+
+	actualTableName := result.TableNames[0]
+	require.Equal(t, tableName, actualTableName)
+
+	// add and query data from the second container
+	value := "test_value"
+	err = addDataToTable(cli2, value)
+	require.NoError(t, err, "data should be added to dynamodb table")
+
+	// read data from the first container
+	queryResult, err := queryItem(cli1, value)
+	require.NoError(t, err, "data should be queried from dynamodb table")
+	require.Equal(t, value, queryResult)
+}
+
+func TestRun_withoutSharedDB(t *testing.T) {
+	ctx := context.Background()
+
+	ctr1, err := tcdynamodb.Run(ctx, image2_2_1)
+	testcontainers.CleanupContainer(t, ctr1)
+	require.NoError(t, err)
+
+	cli, err := getDynamoDBClient(ctr1)
+	require.NoError(t, err, "failed to get dynamodb client handle")
+
+	requireTableExists(t, cli, tableName)
+
+	// create a second container: it should not have the table created in the first container
+	ctr2, err := tcdynamodb.Run(ctx, image2_2_1)
+	testcontainers.CleanupContainer(t, ctr2)
+	require.NoError(t, err)
+
+	// fetch client handle again
+	cli, err = getDynamoDBClient(ctr2)
+	require.NoError(t, err, "failed to get dynamodb client handle")
+
+	// list tables and verify
+
+	result, err := cli.ListTables(context.Background(), nil)
+	require.NoError(t, err, "dynamodb list tables operation failed")
+	require.Empty(t, result.TableNames, "table should not exist after restarting container")
+}
+
+func TestRun_shouldStartWithTelemetryDisabled(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := tcdynamodb.Run(ctx, image2_2_1, tcdynamodb.WithDisableTelemetry())
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+}
+
+func TestRun_shouldStartWithSharedDBEnabledAndTelemetryDisabled(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := tcdynamodb.Run(ctx, image2_2_1, tcdynamodb.WithSharedDB(), tcdynamodb.WithDisableTelemetry())
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+}
+
+func createTable(client *dynamodb.Client) error {
+	_, err := client.CreateTable(context.Background(), &dynamodb.CreateTableInput{
+		TableName: aws.String(tableName),
+		KeySchema: []types.KeySchemaElement{
+			{
+				AttributeName: aws.String(pkColumnName),
+				KeyType:       types.KeyTypeHash,
+			},
+		},
+		AttributeDefinitions: []types.AttributeDefinition{
+			{
+				AttributeName: aws.String(pkColumnName),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+		},
+		BillingMode: types.BillingModePayPerRequest,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addDataToTable(client *dynamodb.Client, val string) error {
```

Comment:
suggestion: this is returning an error only to assert its not one, pass in t to simplify.